src/main/java/programminglife/gui/controller/GraphController.java

```diff
@@ -66,33 +70,48 @@ public void draw(int center, int radius) {
         subGraph.layout();
         long finishTimeLayout = System.nanoTime();
 
-        long startTimeDrawing = System.nanoTime();
+        long startTimeColorize = System.nanoTime();
+        colorize();
+        long finishTimeColorize = System.nanoTime();
 
-        for (DrawableNode drawableNode : subGraph.getNodes().values()) {
-            drawNode(drawableNode);
-            for (DrawableNode child : subGraph.getChildren(drawableNode)) {
-                drawEdge(drawableNode, child);
-            }
-        }
+
+        long startTimeDrawing = System.nanoTime();
+        draw(gc);
         long finishTimeDrawing = System.nanoTime();
 
-        centerOnNodeId(center);
+//        long startHighlight = System.nanoTime();
         highlightNode(center, Color.DARKORANGE);
+        centerOnNodeId(center);
+        System.out.println(this.subGraph.getNodes().get(center).getLocation());
+
+//        Console.println("Time to highlight: " + (System.nanoTime() - startHighlight) / 1000000 + " ms");
 
         long finishTime = System.nanoTime();
         long differenceTimeProgram = finishTime - startTimeProgram;
         long differenceTimeDrawing = finishTimeDrawing - startTimeDrawing;
         long differenceTimeLayout = finishTimeLayout - startLayoutTime;
         long differenceTimeSubGraph = finishTimeSubGraph - startTimeSubGraph;
+        long differenceTimeColorize = finishTimeColorize - startTimeColorize;
         long msDifferenceTimeProgram = differenceTimeProgram / 1000000;
-        long millisecondTimeDrawing = differenceTimeDrawing   / 1000000;
-        long msDifferenceTimeLayout = differenceTimeLayout   / 1000000;
+        long millisecondTimeDrawing = differenceTimeDrawing / 1000000;
+        long msDifferenceTimeLayout = differenceTimeLayout / 1000000;
         long msDifferenceTimeSubGraph = differenceTimeSubGraph / 1000000;
+        long msDifferenceTimeColorize = differenceTimeColorize / 1000000;
         Console.println("time of SubGraph: " + msDifferenceTimeSubGraph);
-        Console.println("Time of Drawing:  " + millisecondTimeDrawing);
         Console.println("Time of layout:  " + msDifferenceTimeLayout);
+        Console.println("Time of Colorize:  " + msDifferenceTimeColorize);
+        Console.println("Time of Drawing:  " + millisecondTimeDrawing);
         Console.println("Time of Total Program:  " + msDifferenceTimeProgram);
+    }
 
+    /**
+     * Method to do the coloring of the to be drawn graph.
+     */
+    private void colorize() {
+        canvas.getGraphicsContext2D().setLineWidth(3.0);
```

Comment:
The name is a bit strange but I think it still has to be done. setLineWidth does get called already in the drawableSegment.colorize() method so I removed it here.